### DISCUSSION TITLE
Flashing kprepublic bm40

### DISCUSSION THREAD

**Discussion by leo0liver** [manna-harbour/miryoku#188] [Q&A] [answered]
Hi,

I have a kprepublic bm40 which is a qmk supported planck knock off.
I've got qmk compiling and flashing the default keymap but im struggling to flash miryoku to it.
Ive merged the miryoku development branch into my qmk repo and then run the command:
`make kprepublic/bm40hsrgb:manna-harbour_miryoku:flash`
From the root directory but that results in the error:
`make[1]: *** No rule to make target 'manna-harbour_miryoku:flash'.  Stop.
Making kprepublic/bm40hsrgb/rev2 with keymap default and target manna-harbour_miryoku:flash    [ERROR]`

I presume this is because the bm40 doesn't support community layouts? Does anyone know if there is any other way I can flash miryoku to my keyboard or will I have to recreate the keymap myself? 

Thanks!

**Comment by manna-harbour** [comment, marked as answer]
Yes, it just needs community layout support enabled. Here's a fix: https://github.com/manna-harbour/qmk_firmware/tree/community-layout-fix-20230130. If you merge that branch you'll be able to build. Here's a sample build run, with RGB disabled due to firmware size: https://github.com/manna-harbour/miryoku_qmk/actions/runs/4039283401.

**Comment by leo0liver** [comment]
Thanks :)
I ended up getting it working :)

What I did was:

1. Add the line `community_layouts": ["planck_mit"]` to the bm40 info.json file
2. Change the layout name from `ortho_4x12_1x2uC` to `planck_mit` in info.json and bm40hsrgb.h.

Then to reduce the size of the firmware I added the line `LTO_ENABLE = yes` to the rules.mk file.
